apps/web/src/hooks/useOutsideClick.ts

```diff
@@ -0,0 +1,28 @@
+import { ReactNode } from '@tanstack/react-router';
+import { RefObject, useEffect } from 'react';
+
+const useOutsideClick = (ref: RefObject<HTMLElement>, onClose: () => void) => {
+  useEffect(() => {
+    const handleClickOutside = (e: MouseEvent) => {
+      if (ref.current && !ref.current.contains(e.target as ReactNode)) {
```

Comment:
p3:
```suggestion
      if (!ref.current || ref.current.contains(e.target as ReactNode)) {
        return;
      }
```

아래 부분도 동일하게 변경해 주세요! 🫡

Reply:
이렇게 변경해야하는 이유를 알려주실 수 있을까요?!
찾아보니 부정로직을 먼저 검사하는 early return 구조가 더 가독성이 좋다.. 이런 이야기가 있는 것 같은데 그런 의도가 맞을까요?